SecretSantaServer/SecretSantaServer/src/Attributes/EmailValidateAttribute.cs

```diff
@@ -0,0 +1,54 @@
+using System.ComponentModel.DataAnnotations;
+
+namespace SecretSantaServer.Attributes;
+
+public class EmailValidateAttribute : ValidationAttribute
+{
+    protected override ValidationResult? IsValid(object? value, ValidationContext validationContext)
+    {
+        if (value is not string email || string.IsNullOrWhiteSpace(email))
+            return ValidationResult.Success;
+
+        if (email.Contains(' '))
+            return new ValidationResult(ErrorMessage ?? "Email must not contain spaces.");
+
+        if (email.Any(c => c > 127))
+            return new ValidationResult(ErrorMessage ?? "Email must contain only Latin letters, digits, and standard symbols.");
+
+        var atCount = email.Count(c => c == '@');
+        if (atCount != 1)
+            return new ValidationResult(ErrorMessage ?? "Email must contain exactly one '@' symbol.");
+
+        var parts = email.Split('@');
+        var localPart = parts[0];
+        var domain = parts[1];
+
+        if (string.IsNullOrEmpty(localPart) || string.IsNullOrEmpty(domain))
+            return new ValidationResult(ErrorMessage ?? "Email is missing local or domain part.");
+
+        if (!domain.Contains('.'))
+            return new ValidationResult(ErrorMessage ?? "Domain must contain a dot (e.g. @gmail.com).");
```

Comment:
```suggestion
            return new ValidationResult(ErrorMessage ?? "Domain must contain a dot (e.g. gmail.com).");
```

`@gmail.com` - `@` char is not a pert of Domain